protocol-designer/src/pages/CreateNewProtocolWizard/utils.tsx

```diff
@@ -42,6 +46,35 @@ export const getNumOptions = (length: number): DropdownOption[] => {
   }))
 }
 
+// Note (1/31/25): at this moment, users allow to set one about thermocycler and need to count 2
+interface ModuleCounts {
+  magneticBlockCount: number
+  heaterShakerCount: number
+  temperatureCount: number
+  plateReaderCount: number
+}
+
+const countModules = (modules: WizardFormState['modules']): ModuleCounts => {
+  const magneticBlockCount = Object.values(modules || {}).filter(
+    module => module.type === MAGNETIC_BLOCK_TYPE
+  ).length
+  const heaterShakerCount = Object.values(modules || {}).filter(
+    module => module.type === HEATERSHAKER_MODULE_TYPE
+  ).length
+  const temperatureCount = Object.values(modules || {}).filter(
+    module => module.type === TEMPERATURE_MODULE_TYPE
+  ).length
+  const plateReaderCount = Object.values(modules || {}).filter(
+    module => module.type === ABSORBANCE_READER_TYPE
+  ).length
+  return {
+    magneticBlockCount,
+    heaterShakerCount,
+    temperatureCount,
+    plateReaderCount,
+  }
+}
```

Comment:
maybe for efficiency we can do something like this:
```suggestion
const countModules = (modules: WizardFormState['modules']): ModuleCounts =>
  Object.values(modules || {}).reduce<ModuleCounts>(
    (acc, { type }) => {
      if (type === MAGNETIC_BLOCK_TYPE) {
        return { ...acc, magneticBlockCount: acc.magneticBlockCount + 1 }
      } else if (type === HEATERSHAKER_MODULE_TYPE) {
        return { ...acc, heaterShakerCount: acc.heaterShakerCount + 1 }
      } else if (type === TEMPERATURE_MODULE_TYPE) {
        return { ...acc, heaterShakerCount: acc.temperatureCount + 1 }
      } else if (type === ABSORBANCE_READER_TYPE) {
        return { ...acc, heaterShakerCount: acc.plateReaderCount + 1 }
      }
      return acc
    },
    {
      magneticBlockCount: 0,
      heaterShakerCount: 0,
      temperatureCount: 0,
      plateReaderCount: 0,
    }
  )
```

Reply:
or even 
```suggestion
const typeMappedToCountKey: Partial<Record<ModuleType, keyof ModuleCounts>> = {
  [MAGNETIC_BLOCK_TYPE]: 'magneticBlockCount',
  [HEATERSHAKER_MODULE_TYPE]: 'heaterShakerCount',
  [TEMPERATURE_MODULE_TYPE]: 'temperatureCount',
  [ABSORBANCE_READER_TYPE]: 'plateReaderCount',
}

const countModules = (modules: WizardFormState['modules']): ModuleCounts =>
  Object.values(modules || {}).reduce<ModuleCounts>(
    (acc, { type }) => {
      const countKey = typeMappedToCountKey[type] ?? null
      return countKey != null ? { ...acc, [countKey]: acc[countKey] + 1 } : acc
    },
    {
      magneticBlockCount: 0,
      heaterShakerCount: 0,
      temperatureCount: 0,
      plateReaderCount: 0,
    }
  )
```
may be easier to just check the key in the if / else blocks though